File: src/main/java/InvertBinaryTree/GGOBOND/Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return root
        left_tr = root.left
        right_tr = root.right
        root.left = self.invertTree(right_tr)
        root.right = self.invertTree(left_tr)
        
        return root

File: src/main/java/InvertBinaryTree/muzimusi/InvertBinaryTree.java
package InvertBinaryTree.muzimusi;


//  Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

public class InvertBinaryTree {
    public TreeNode invertTree(TreeNode root) {
        if (root == null)
            return root;
        // 递归翻转左子树
        TreeNode leftRoot = invertTree(root.left);
        // 递归翻转右子树
        TreeNode rightRoot = invertTree(root.right);
        // 交换左右子树
        root.left = rightRoot;
        root.right = leftRoot;
        return root;
    }
}
